nautobot/core/views/mixins.py

```diff
@@ -499,9 +520,22 @@ def get_queryset(self):
         queryset = super().get_queryset()
         return queryset.restrict(self.request.user, self.get_action())
 
+    def get_action_map(self):
+        """
+        Return a mapping of action names to their corresponding permissions.
+        This is used to determine the permissions required for each action.
+        """
+        return PERMISSIONS_ACTION_MAP
+
     def get_action(self):
         """Helper method for retrieving action and if action not set defaulting to action name."""
-        return PERMISSIONS_ACTION_MAP.get(self.action, self.action)
+        action_map = self.get_action_map()
+        if self.action in action_map:
+            # If the action is in the action_map, return the mapped permission
+            return action_map[self.action]
+        elif self.custom_view_base_action:
+            return self.custom_view_base_action
```

Comment:
Intuitively I would think that `custom_view_base_action` would take priority over the default behavior, not the other way around?